### DISCUSSION TITLE
请问Anaconda环境中加密后，相同环境无法执行加密后的py文件

### DISCUSSION THREAD

**Discussion by yunkai-jia** [dashingsoft/pyarmor#1279] [Q&A]
我在anaconda环境中使用 pyarmor gen foo.py进行加密了，加密成功了，如下：
(JiaYunkai_code) jiayunkai@jiayunkai-3 pyarmor加密-test % pyarmor gen foo.py
INFO     Python 3.9.13
INFO     Pyarmor 8.1.7 (pro), 005013, non-profits
INFO     search inputs ...86_64
INFO     find script foo.py
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'darwin.x86_64'}
INFO     write dist/pyarmor_runtime_005013/pyarmor_runtime.so
INFO     patch runtime file
INFO     sign runtime file
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "foo"
INFO     obfuscating file foo.py
INFO     write dist/foo.py
INFO     obfuscate scripts OK

但是加密成功后，我在同样的加密环境中进行运行，出现下面的结果，不好意思啦？
“”“
，/opt/anaconda3/envs/algo/bin/python /Users/jiayunkai/PycharmProjects/pyarmor加密/dist/foo.py

进程已结束，退出代码为 139 (interrupted by signal 11: SIGSEGV)
”“”

**Comment by jondy** [comment]
请升级到最新版本 8.2.2，同时参阅文档中的相关说明。

**Comment by jondy** [comment]
关于 Apple M1 下面的 segment fault 已经报告了很多类似issue了，可以搜索参考一下，另外中文的文档中也有专门说明了，请学会自己解决问题。

**Comment by yunkai-jia** [comment]
现在pyarmor已经更新为8.2.2
INFO     Python 3.9.13
INFO     Pyarmor 8.2.2 (pro), 005013, non-profits(TBD)
INFO     Platform darwin.x86_64
INFO     search inputs ...

**Comment by yunkai-jia** [comment]
也看了文档中6.3所说的Apple M1 上的 Segment fault，我执行了上述操作，但似乎还出现segmentation fault：
(JiaYunkai_code) jiayunkai@jiayunkai-3 pyarmor加密-test % codesign -s - -f dist/pyarmor_runtime_005013/pyarmor_runtime.so
dist/pyarmor_runtime_005013/pyarmor_runtime.so: replacing existing signature
(JiaYunkai_code) jiayunkai@jiayunkai-3 pyarmor加密-test % python dist/foo.py                                             
zsh: segmentation fault  python dist/foo.py
这种情况是什么原因呢？

**Comment by jondy** [comment]
网上搜一下吧，一般解决方案就是 
1.  codesign
2. 使用 `otool -L dist/pyarmor_runtime_005013/pyarmor_runtime.so` 检查依赖库，如果没有系统依赖库的话使用 install_name_tool 修改依赖库。
3. 看一下 Apple 的 segment fault 日志，根据提示的错误信息搜索网络找解决方案
4. 确保 `dist/pyarmor_runtime_005013/pyarmor_runtime.so` 链接的 Python 库 和 python 命令是一致的，如果系统上装了多个 Python 的话
总之，这是系统配置问题，无法装载扩展模块 pyarmor_runtime.so 造成的。

**Comment by jondy** [comment]
我刚刚在文档中增加了具体的一些解决方案，请参阅
https://pyarmor.readthedocs.io/zh/latest/questions.html#apple-segment-fault
